sdk/pinocchio/src/sysvars/clock.rs

```diff
@@ -65,3 +76,67 @@ pub const DEFAULT_MS_PER_SLOT: u64 = 1_000 * DEFAULT_TICKS_PER_SLOT / DEFAULT_TI
 impl Sysvar for Clock {
     impl_sysvar_get!(sol_get_clock_sysvar);
 }
+
+impl Clock {
+    /// The length of the `Clock` sysvar account data.
+    pub const LEN: usize = 8 + 8 + 8 + 8 + 8;
+
+    /// Return a `Clock` from the given account info.
+    ///
+    /// This method performs a check on the account info key.
+    #[inline]
+    pub fn from_account_info(account_info: &AccountInfo) -> Result<Ref<Clock>, ProgramError> {
+        if account_info.key() != &CLOCK_ID {
+            return Err(ProgramError::InvalidArgument);
+        }
+        Ok(Ref::map(account_info.try_borrow_data()?, |data| unsafe {
+            Self::from_bytes_unchecked(data)
+        }))
+    }
+
+    /// Return a `Clock` from the given account info.
+    ///
+    /// This method performs a check on the account info key, but does not
+    /// perform the borrow check.
+    ///
+    /// # Safety
+    ///
+    /// The caller must ensure that it is safe to borrow the account data – e.g., there are
+    /// no mutable borrows of the account data.
+    #[inline]
+    pub unsafe fn from_account_info_unchecked(
+        account_info: &AccountInfo,
+    ) -> Result<&Self, ProgramError> {
+        if account_info.key() != &CLOCK_ID {
+            return Err(ProgramError::InvalidArgument);
+        }
+        Ok(Self::from_bytes_unchecked(
+            account_info.borrow_data_unchecked(),
+        ))
+    }
+
+    /// Return a `Clock` from the given bytes.
+    ///
+    /// This method performs a length validation. The caller must ensure that `bytes` contains
+    /// a valid representation of `Clock`.
```

Comment:
We are missing the `# Safety` heading here. Let's move the second sentence from the line above to it:
```rust
/// # Safety
///
/// The caller must ensure that `bytes` contains a valid representation of `Clock`.
```